File: app/src/main/java/team/triplog/presentation/adapter/TripAlbumAdapter.java
package team.triplog.presentation.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

import team.triplog.R;

public class TripAlbumAdapter extends RecyclerView.Adapter<TripAlbumAdapter.ViewHolder> {
    private Context context;
    private ArrayList<String> items;
    private OnItemClickListener onItemClickListener;

    public TripAlbumAdapter(
            Context context,
            ArrayList<String> images,
            OnItemClickListener onItemClickListener
    ) {
        this.context = context;
        this.items = images;
        this.onItemClickListener = onItemClickListener;
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                LayoutInflater
                        .from(parent.getContext())
                        .inflate(R.layout.item_trip_album, parent, false),
                onItemClickListener
        );
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        String imagePath = items.get(position);

        Glide.with(context)
                .load(imagePath)
                .thumbnail(0.1f)
                .into(holder.viewAlbum);
    }

    public void setItems(ArrayList<String> items) {
        this.items = items;
        notifyDataSetChanged();
    }

    static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        OnItemClickListener onItemClickListener;
        ImageView viewAlbum;

        ViewHolder(View view, OnItemClickListener onItemClickListener) {
            super(view);

            this.onItemClickListener = onItemClickListener;
            viewAlbum = view.findViewById(R.id.view_album);

            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            onItemClickListener.onItemClick(view, getAdapterPosition());
        }
    }

    public interface OnItemClickListener {
        void onItemClick(View view, int position);
    }
}

File: app/src/main/java/team/triplog/presentation/trip/add/adapter/TripAlbumAdapter.kt
package team.triplog.presentation.trip.add.adapter

import android.content.Context
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import team.triplog.presentation.trip.add.viewholder.TripAlbumViewHolder
import java.util.*


/**
 * @author mjkim
 * @since 2021/06/06
 **/
class TripAlbumAdapter(
    private val context: Context,
    private var items: ArrayList<String>,
    private val onItemClickListener: OnItemClickListener
) : RecyclerView.Adapter<TripAlbumViewHolder>() {
    override fun getItemCount(): Int = items.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TripAlbumViewHolder {
        return TripAlbumViewHolder.getViewHolder(parent)
    }

    override fun onBindViewHolder(holder: TripAlbumViewHolder, position: Int) {
        val item = items[position]

        holder.bind(context, item, onItemClickListener)

    }

    fun setItems(items: ArrayList<String>) {
        this.items = items
        notifyDataSetChanged()
    }

    interface OnItemClickListener {
        // TODO : 현재는 item 값을 String 으로 받고있지만, 추후 사진 관련 정보를 담는 model 로 치환 예정.
        fun onItemClick(item: String)
    }
}